### PR TITLE
Correct the expected result of the given code

### LINKED ISSUE #18082: Wrong Promise.prototype.finally() description
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

### What specific section or headline is this issue about?

Description

### What information was incorrect, unhelpful, or incomplete?

There are 2 paragraphs in the [Description](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally#description) where I found to be misleading due to the fact that I tried to execute those expressions and the output is not as mentioned:
image
  

### What did you expect to see?

- When I executed `Promise.reject(3).then(() => {}, () => 88)`, it returns a fulfilled promise with the value `88` (**not** "_return a rejected promise with the reason `88`_" as mentioned in the section):
image

Which matches the [return value of the `Promise.prototype.then()` method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then#return_value):
image

- The `Promise.reject(3).finally(() => throw 99)` actually throws an uncaught exception:
image

Which I believe just a typo. Can be fixed by adding a pair of curly braces: `Promise.reject(3).finally(() => {throw 99});`

- The execution of `Promise.reject(3).finally(() => Promise.reject(99))` actually returns a rejected promise with the reason `99`:
image

Which abides by the [return value of the `Promise.prototype.finally()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally#return_value) method:
image

- Assuming the typo from `Promise.reject(3).finally(() => throw 99)` is fixed (because if it were intended to be syntactically wrong, the author should've mentioned that), the phrasing with "either...or" isn't the right choice of words, because they **BOTH** will return a promise that is rejected and has a value of `99`.

### Do you have any supporting links, references, or citations?

Already listed above.

### Do you have anything more you want to share?

I'm still learning Promise and its related topics so I'm unable to formulate a definitive example for that case. I'd like the reviewers to have their insights on this and maybe come up with a better code sample to ease in the reading process.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/javascript/reference/global_objects/promise/finally`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/promise/finally/index.md
* Last commit: https://github.com/mdn/content/commit/4e4ec5dbe15ae4883201057aaab8833dd33b7a05
* Document last modified: 2022-06-30T11:02:34.000Z

</details>

### PR DESCRIPTION
**Pull request by hiteshvijayhv**
In this commit I corrected the output of the below code.
```
Promise.reject(3).then(() => {}, () => 88)
```
as this returns a fulfilled/resolved promise

#### Summary

Correcting the expected output of the code.

#### Supporting details

Fixing the doc [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally)

#### Related issues

Fixes #18082
#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [X] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 1]
```diff
@@ -64,10 +64,10 @@ The `finally()` method is very similar to calling
     `Promise.resolve(2).finally(() => 77)` will return a
     new resolved promise with the result `2`.
   - Similarly, unlike `Promise.reject(3).then(() => {}, () => 88)`
-    (which will return a rejected promise with the reason `88`),
+    (which will return a resolved promise with the reason `88`),
```
Comment (HelpMe-Pls) [inline files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:67]: IMHO it should be rephased as " which will return a **_fulfilled_** promise with the _**value**_ `88`"

**Comment by Josh-Cena** [comment]
The description section can benefit from some more thorough refactor. I'll commit to your branch.

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 2]
```diff
@@ -64,10 +64,10 @@ The `finally()` method is very similar to calling
     `Promise.resolve(2).finally(() => 77)` will return a
```
Comment (HelpMe-Pls) [inline files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:64]: For consistency, I think this should be rephrased as "return a new fulfilled promise with the value `2`:
image

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md [hunk 3]
```diff
@@ -64,10 +64,10 @@ The `finally()` method is very similar to calling
...
     new resolved promise with the result `2`.
   - Similarly, unlike `Promise.reject(3).then(() => {}, () => 88)`
-    (which will return a rejected promise with the reason `88`),
+    (which will return a resolved promise with the reason `88`),
     `Promise.reject(3).finally(() => 88)` will return a rejected promise
     with the reason `3`.  
-  - But, either `Promise.reject(3).finally(() => throw 99)` or
+  - But, either `Promise.reject(3).finally(() => {throw 99})` or
```
Comment (HelpMe-Pls) [inline files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:70]: This should be rephrased from "either...or..." to "Both...and...":
"Both `Promise.reject(3).finally(() => {throw 99})` and `Promise.reject(3).finally(() => Promise.reject(99))` will return a rejected promise with the reason `99`.

**Review by HelpMe-Pls — COMMENTED** [review]
Please have a look at my suggestions (in case there're no better examples for this particular case).

File: files/en-us/web/javascript/reference/global_objects/promise/finally/index.md (hunk 3 above)
Comment (hiteshvijayhv) [reply files/en-us/web/javascript/reference/global_objects/promise/finally/index.md:70]: > This should be rephrased from "either...or..." to "Both...and...": "Both `Promise.reject(3).finally(() => {throw 99})` and `Promise.reject(3).finally(() => Promise.reject(99))` will return a rejected promise with the reason `99`.

Yes will fix this as well

**Comment by teoli2003** [comment]
Hi! Not sure to understand why this has been closed. Any reason?

**Pull request closed without merging**